### PR TITLE
Added Moderation with Muting, Banning, Warning

### PR DESCRIPTION
**Pull request by Jay-Madden**

### CODE REVIEW

File: bot/cogs/help_cog.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ async def send_group_help(self, ctx, command: commands.Group):
 
         await ctx.send(embed=embed)
 
-    async def send_command_help(self, ctx, command: commands.Command):
+    async def send_command_help(self, ctx, command: ext.ClemBotCommand):
```
Comment (mp7786) [inline bot/cogs/help_cog.py:62]: What was the purpose of building a class instead of using commands.Command?

Comment (Jay-Madden) [reply bot/cogs/help_cog.py:62]: ext.ClemBotCommand is clembots extension class to allow for the short_help long_help decorators we use

Comment (mp7786) [reply bot/cogs/help_cog.py:62]: Oh that makes sense.

File: bot/cogs/moderation_cog/ban_cog.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                                             DesignatedChannels.moderation_log,
+                                             ctx.guild.id,
+                                             embed)
+
+        await self.bot.messenger.publish(Events.on_bot_ban,
+                                         guild=ctx.guild,
+                                         author=ctx.author,
+                                         subject=subject,
+                                         reason=reason)
+
```
Comment (mp7786) [inline bot/cogs/moderation_cog/ban_cog.py:56]: Would it be a good idea to wrap this with an try-except as you currently send the ban messages before the ban has actually occurred?

Comment (Jay-Madden) [reply bot/cogs/moderation_cog/ban_cog.py:56]: we want the error to bubble up to the global handler, no need to handle it. if the ban fails we want everything to fail

Comment (mp7786) [reply bot/cogs/moderation_cog/ban_cog.py:56]: I also meant that you send the ban message to the banned user before the ban has been attempted.

**Pull request merged**